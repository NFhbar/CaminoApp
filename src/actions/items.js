import uuid from 'uuid';

// ADD_ITEM
export const addItem = (
  {
    description = '',
    note = '',
    amount = 0
  } = {}
) => ({
  type: 'ADD_ITEM',
  item: {
    id: uuid(),
    description,
    note,
    amount,
    sales: Math.floor(Math.random()*Math.floor(100)),
    sales2: Math.floor(Math.random()*Math.floor(100)),
    sales3: Math.floor(Math.random()*Math.floor(100)),
    sales4: Math.floor(Math.random()*Math.floor(100))
  }
});

// REMOVE_ITEM
export const removeItem = ({ id } = {}) => ({
  type: 'REMOVE_ITEM',
  id
});

// EDIT_ITEM
export const editItem = (id, updates) => ({
  type: 'EDIT_ITEM',
  id,
  updates
});
