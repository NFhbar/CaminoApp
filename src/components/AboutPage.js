import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const AboutPage = () => (
  <div className="row">
    <div className="col-sm-12">
      <p>From About</p>
    </div>
  </div>
);

export default AboutPage;
