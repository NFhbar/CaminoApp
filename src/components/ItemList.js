import React from 'react';
import { connect } from 'react-redux';
import ItemListItem from './ItemListItem';
import selectItems from '../selectors/items';

export const ItemList = (props) => (
  <div>
    {
      props.items.length === 0 ? (
        <p>No Items</p>
      ) : (
        props.items.map((item) => {
          return <ItemListItem key={item.id} {...item}/>
        })
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    items: selectItems(state.items, state.filters)
  };
};

export default connect(mapStateToProps)(ItemList);
