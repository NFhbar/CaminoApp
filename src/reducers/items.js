// Items Reducer

const itemsReducerDefaultState = [];

export default (state = itemsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_ITEM':
      return [
        ...state,
        action.item
      ];
    case 'REMOVE_ITEM':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_ITEM':
      return state.map((item) => {
        if(item.id === action.id){
          return {
            ...item,
            ...action.updates
          };
        } else {
          return item;
        };
      });
    default:
      return state;
  }
};
