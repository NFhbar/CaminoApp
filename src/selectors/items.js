// Get visible items
export default (items, {text, sortBy}) => {
  return items.filter((item) => {
    const textMatch = item.description.toLowerCase().includes(text.toLowerCase());

    return textMatch;
  }).sort((a, b) => {
    if (sortBy === 'amount') {
      return a.amount < b.amount ? 1: -1;
    } else if (sortBy === 'sales') {
      return a.sales < b.sales ? 1: -1;
    }
  });
};
