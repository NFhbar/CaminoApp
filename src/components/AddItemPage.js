import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm'
import { addItem } from '../actions/items';

export class AddItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.addItem(item);
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row card-body">
        <div className="col-sm-12">
          <h3>Add an item to your store!</h3>
          <ItemForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item))
});

export default connect(undefined, mapDispatchToProps)(AddItemPage);
