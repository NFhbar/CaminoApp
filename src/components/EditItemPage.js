import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm';
import { editItem, startRemoveItem } from '../actions/items';

export class EditItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.editItem(this.props.item.id, item);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveItem({ id: this.props.item.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row" id="edit-item-page">
        <div className="col-sm-12">
          <ItemForm
            item={this.props.item}
            onSubmit={this.onSubmit}
          />
          <button type="submit" id="btn-remove" className="btn btn-md btn-danger" onClick={this.onRemove}>Remove Item</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  item: state.items.find((item) => item.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editItem: (id, item) => dispatch(editItem(id, item)),
  startRemoveItem: (data) => dispatch(startRemoveItem(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditItemPage);
