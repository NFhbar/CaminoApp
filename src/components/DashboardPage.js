import React from 'react';
import { row, column } from 'reactstrap';
import ItemList from './ItemList';
import ItemPlot from './ItemPlot';
import ItemListFilters from './ItemListFilters';

const DashboardPage = () => (
  <div className="row card-body">
    <div className="col-sm-12">
      <ItemListFilters />
      <ul className="list-group">
        <li className="list-group-item disabled">
          <div className="row">
            <div className="col-sm-4">
              Item
            </div>
            <div className="col-sm-4">
              Price
            </div>
            <div className="col-sm-4">
              Items sold
            </div>
          </div>
        </li>
        <ItemList />
      </ul>
      <ItemPlot />
    </div>
  </div>
);

export default DashboardPage;
