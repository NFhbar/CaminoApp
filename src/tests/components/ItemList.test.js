import React from 'react';
import { shallow } from 'enzyme';
import { ItemList } from '../../components/ItemList';
import items from '../fixtures/items';

test('should render ItemList with items', () => {
  const wrapper = shallow(<ItemList items={items} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ItemList with empty message', () => {
  const wrapper = shallow(<ItemList items={[]} />);
  expect(wrapper).toMatchSnapshot();
});
