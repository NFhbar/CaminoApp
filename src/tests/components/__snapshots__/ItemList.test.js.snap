// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ItemList with empty message 1`] = `
<li
  className="list-group-item"
>
  <div
    className="alert alert-primary"
    role="alert"
  >
    Add some items to get started!
  </div>
</li>
`;

exports[`should render ItemList with items 1`] = `
<li
  className="list-group-item"
>
  <ItemListItem
    amount={500}
    description="Pant"
    id="1"
    key="1"
    note=""
    sales={10}
    sales2={20}
    sales3={30}
    sales4={40}
  />
  <ItemListItem
    amount={1000}
    description="Shirt"
    id="2"
    key="2"
    note=""
    sales={1}
    sales2={2}
    sales3={3}
    sales4={4}
  />
  <ItemListItem
    amount={1500}
    description="Coat"
    id="3"
    key="3"
    note=""
    sales={5}
    sales2={10}
    sales3={15}
    sales4={20}
  />
</li>
`;
